car/Makefile

```diff
@@ -1,20 +1,25 @@
-TESTSUITE_DIR := /testsuite
+API_HOST := localhost
+
+TESTSUITE_DIR := ./testsuite
 
 TESTS := $(addprefix $(TESTSUITE_DIR)/test,1 2 3 4 5 6 7 8 9)
 
-all:
-	gcc -o gps_uploader -g -fsanitize=undefined -fno-sanitize-recover src/gps_uploader.c
+.PHONY: all
+all: gps_uploader
+
+gps_uploader:
+	gcc -o $@ -g -fsanitize=undefined -fno-sanitize-recover src/gps_uploader.c
 
-test: $(TESTS)
+test: $(TESTS) gps_uploader
 	@for test in $^; do \
 		echo "- Running $$test..."; \
 		./gps_uploader  $$test; \
 	done
 
-upload: $(TESTS)
+upload: $(TESTS) gps_uploader
 	@for test in $^; do \
 		echo "- Running $$test..."; \
-		./gps_uploader  $$test https://api:8443/location; \
```

Comment:
Any reason we change this here? `api` is read by `docker-compose.yaml`, so points to wherever the API container is running (almost always localhost, but could change if we end up using codespaces, which would make this brittle). I see you are overwriting in the car service, but I am not sure this propagates to the other services. If we're going to change it we should change it everywhere.